### PR TITLE
feat(broker): make embedded gateway switchable

### PR DESCRIPTION
**Pull request by ezolnbl-dpc**

### CODE REVIEW

**Comment by salaboy** [comment]
Hi @ezolnbl. Thanks for your PR.

I'm waiting for a [zeebe-io](https://github.com/orgs/zeebe-io/people) or [salaboy](https://github.com/orgs/salaboy/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/salaboy/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-cluster-helm).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by salaboy** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **npepinpe**
You can assign the PR to them by writing `/assign @npepinpe` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-cluster-helm).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](zeebe-io/zeebe-cluster-helm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by salaboy** [comment]
@npepinpe @Zelldon I am ok with this change.. can you please check it? it is blocking this user .. we need to wait for the CLA to be signed as well. 
I am not super happy with the name.. but I cannot come up with a better name for the variable..

File: README.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ This functionality is in beta and is subject to change. The design and code is l
 | `ioThreadCount`                 | Set the Zeebe Cluster IO thread count                                                                                                                                | `2`  
 | `zeebeCfg`                 | Can be used to set several zeebe configuration options.                                                                                                                                | `null`
 | `gatewayMetrics`                 | Enables the exporting of the gateway prometheus metrics                                                                                                                                | `false`
+| `brokerEmbeddedGatewayEnable`                 | Enables the embedded broker gateway                                                                                                                                | `false`
```
Comment (npepinpe) [inline README.md:45]:
```suggestion
| `embeddedGatewayEnabled`                 | Enables the embedded gateway in all brokers                                                                                                                                | `false`
```

File: zeebe-cluster/templates/statefulset.yaml [hunk 2]
```diff
@@ -51,7 +51,7 @@ spec:
         - name: ZEEBE_BROKER_THREADS_IOTHREADCOUNT
           value: {{ .Values.ioThreadCount  | quote }}
         - name: ZEEBE_BROKER_GATEWAY_ENABLE
-          value: "false"
+          value: {{ .Values.brokerEmbeddedGatewayEnable | default "false" | quote }}
```
Comment (npepinpe) [inline zeebe-cluster/templates/statefulset.yaml:54]:
```suggestion
          value: {{ .Values.embeddedGatewayEnabled | default "false" | quote }}
```

**Comment by falko** [comment]
+1 for the new name.

Couldn't we just use an additional custom label that is assigned to any gateway nodes regardless of embedded or standalone? That would even allow a combination of standardlone and embedded gateways if that makes any sense. The standard component labels could remain as they are.

**Comment by salaboy** [comment]
My recommendation as always is to keep the topology fixed.. meaning.. the charts should give you the recommended setup.. and in my head at least that’s with the standalone gateway. 
I am worried of making things too complicated without clear advantages. I’ve proposed before to have a separate charts, for the gateway and the brokers.. that might be an alternative approach.

File: README.md (hunk 1 above)
Comment (salaboy) [reply README.md:45]: +1 @ezolnbl can you please add @npepinpe suggestions?

File: zeebe-cluster/templates/statefulset.yaml (hunk 2 above)
Comment (salaboy) [reply zeebe-cluster/templates/statefulset.yaml:54]: +1 @ezolnbl can you please add @npepinpe suggestions?

**Comment by salaboy** [comment]
@npepinpe @falko @Zelldon any new decisions/thoughts on this one?

**Comment by npepinpe** [comment]
I think it's a good feature to have, to be able to use the embedded gateway, especially during development phase/evaluation. So I would vote to have it possible to switch between both, but if it increases the complexity a lot and raises the maintenance effort for you, then I understand if you want to keep it out :slightly_smiling_face: If I have to pick only one (standalone or embedded), then definitely standalone.

**Comment by Zelldon** [comment]
This PR seems to be outdated and the consensus was not reached. Sorry for the inconvenience. If you still interested in the changes feel free to reopen a new PR which is rebased to the latest state. It might make sense to create an issue first to discuss what the expected changes are.

**Pull request closed without merging**